src/contributing/style_guide.md

```diff
@@ -0,0 +1,85 @@
+
+Note that "data engineers" (the maintainers of the Mozilla data platform and tools) are *not* the target audience for DTMO, though they may find some resources here helpful in the course of the work. If something is *only* of interest to a data engineer, it probably belongs elsewhere: see the note below on [implementation-specific articles](#implementation-specific-articles).
+
+## What to write
+
+There are three different types of documentation that are useful as part of a site like DTMO:
+
+* Introductory material: Material intended to help people get their bearings with Mozilla's telemetry system. A set of these articles form the [Getting Started](../concepts/getting_started.html) section on this site.
+* Tutorials & Cookbooks: Instructions on how to perform specific tasks using Mozilla's data platform. The focus here is on how to do things, rather than on what they are.
+* Reference: Reference material on either [datasets](../datasets/reference.md) or the [data platform itself](../reference/index.md). The focus is on describing how things work or what they do, rather than how to use them.
+
+In general, the most useful documentation for newcomers is usually a cookbook or tutorial as [they often don't know where to begin](https://stevelosh.com/blog/2013/09/teach-dont-tell/). For more advanced users, reference material may be more useful.
+
+## What *not* to write
+
+There are a few types of documentation that are less appropriate for a general-purpose reference like DTMO.
+
+### Implementation-specific articles
+
+Any articles that are specific to the implementation of particular data systems or tools should be published alongside the system or tool itself. Examples of this might include:
+
+* Usage guides for command-line tools
+* In-depth documentation on architecture and design choices (beyond a general overview)
+* A general usage guide for [specific data tool](../tools/interfaces.md)
+
+In the past, this type of documentation has gone out of date quickly as people update the implementation while forgetting to update what has been written here. Vast amounts of implementation detail can also be overwhelming to people who just want to get an answer to a data-related question.
+
+That said, it can sometimes be useful to provide a general overview of something here while saving the details for site-specific documentation. A good example of this is the [gcp-ingestion](https://mozilla.github.io/gcp-ingestion/) documentation: while we maintain [a high-level description of the data pipeline here](../concepts/pipeline/gcp_data_pipeline.md), details on the Beam-specific implementation are stored alongside the source on GitHub.
```

Comment:
Sometimes you can provide a general overview of an issue while saving the details for site-specific documentation.